src/convert.rs

```diff
@@ -10,49 +11,51 @@ use {Alpha, Rgb, Luma, Xyz, Yxy, Lab, Lch, Hsv, Hwb, Hsl, Color};
 ///from_luma for Rgb is implemented directly. The from for the same color must override
 ///the default. For example, from_rgb for Rgb will convert via Xyz which needs to be overridden
 ///with self to avoid the unnecessary converison.
-pub trait FromColor<T>: Sized
+pub trait FromColor<Wp = D65, T = f32>: Sized
     where T: Float,
+        Wp: WhitePoint<T>
 {
     ///Convert from XYZ color space
-    fn from_xyz(Xyz<T>) -> Self;
+    fn from_xyz(Xyz<Wp, T>) -> Self;
 
     ///Convert from Yxy color space
-    fn from_yxy(inp: Yxy<T>) -> Self {
-        Self::from_xyz(inp.into_xyz())
+    fn from_yxy(inp: Yxy<Wp, T>) -> Self {
+        let xyz = <Xyz<Wp, T> as FromColor<Wp,T>>::from_yxy(inp);
+        Self::from_xyz(xyz)
```

Comment:
Did you mean to comment on the `Color` enum comment? I think it's enough to change it back to use `new` for now. It won't stop anyone from using it with any other white point, by converting.